units/cache_historical_test_data.go

```diff
@@ -0,0 +1,289 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+
+	"github.com/evergreen-ci/evergreen/model/stats"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	cacheHistoricalTestDataName = "cache-historical-test-data"
+	maxSyncDuration             = time.Hour * 24 * 7 // one week
+)
+
+func init() {
+	registry.AddJobType(cacheHistoricalTestDataName,
+		func() amboy.Job { return makeCacheHistoricalTestDataJob() })
+}
+
+type cacheHistoricalTestDataJob struct {
+	ProjectId string `bson:"project_id" json:"project_id" yaml:"project_id"`
+	job.Base  `bson:"job_base" json:"job_base" yaml:"job_base"`
+}
+
+type dailyStatsRollup map[time.Time]map[string][]string
+
+type generateStatsFn func(string, string, time.Time, []string, time.Time) error
```

Comment:
should we have a type for the arguments to this function. the use of this signature is unclear.

Reply:
I added names to the arguments to help clarify the signature. The functions being called are in another package, so switching to using a struct for the arguments would involve changing that package or creating wrapper for the functions.